trackintel/io/dataset_reader.py

```diff
@@ -253,7 +258,13 @@ def geolife_add_modes_to_triplegs(
     if len(tpls_id_mode_list) == 0:
         tpls["mode"] = np.nan
     else:
-        tpls_id_mode = pd.DataFrame(tpls_id_mode_list).set_index("id")
+        tpls_id_mode = pd.DataFrame(tpls_id_mode_list)
+
+        # chose label with highest overlap
+        tpls_id_mode = tpls_id_mode.sort_values(by=["id", "ratio"])
+        # keep last (df sorted ascending)
+        tpls_id_mode = tpls_id_mode.drop_duplicates(subset="id", keep="last").set_index("id")
+
```

Comment:
I assume it's on purpose that the `ratio` column is not dropped, so it is part of the tpls dataframe that is returned? Maybe it should be renamed then, e.g. to `ratio_mode_overlap`, because it could be confusing to have a column `ratio` in the output of a function that's supposed to assign a mode.

Reply:
dropped